### PR TITLE
Session timeout implementation in directory functionalities

### PR DESCRIPTION
**Pull request by godwinamila**

### CODE REVIEW

File: jaggeryapps/dashboard/directories/js/directory.js [hunk 1]
```diff
@@ -69,15 +93,47 @@ function getDirectories() {
...
             }
-            if (directoryList != null && directoryList.constructor !== Array) {
-                var arr = [];
-                arr[0] = directoryList;
-                directoryList = arr;
-            }
-            drawList();
+
+
+//            if (data) {
```
Comment (dinusha-dilrukshi) [inline jaggeryapps/dashboard/directories/js/directory.js]: Remove commented code.

File: jaggeryapps/dashboard/directories/js/directory.js [hunk 2]
```diff
@@ -34,7 +34,31 @@ function addOrUpdateUserDirectory() {
...
         .done(function (data) {
-            window.location.href = DIRECTORY_LIST_PATH;
+
+            var resp = $.parseJSON(data);
+            if (resp.success == true) {
+                window.location.href = DIRECTORY_LIST_PATH;
+            } else {
+
+                if (typeof resp.reLogin != 'undefined' && resp.reLogin == true) {
+                    window.top.location.href = window.location.protocol + '//' + serverUrl + '/dashboard/logout.jag';
```
Comment (dinusha-dilrukshi) [inline jaggeryapps/dashboard/directories/js/directory.js]: Can we define a config property for context (i.e dashboard) and not have hard coded dashboard preferences.

Comment (godwinamila) [inline jaggeryapps/dashboard/directories/js/directory.js]: Fex

File: jaggeryapps/dashboard/directories/js/directory.js (hunk 1 above)
Comment (godwinamila) [inline jaggeryapps/dashboard/directories/js/directory.js]: Removed

**Pull request merged**